### PR TITLE
Prompt user to select networks upon creating/importing new wallet

### PR DESCRIPTION
**Pull request by justindg**

### CODE REVIEW

**Review by JamesSmartCell — CHANGES_REQUESTED** [review]
Issues found:

- After clicking "create New Wallet", app goes to the "name wallet" screen but it's not clear why or what user is supposed to do here (I was initially confused about what was meant to happen here. If we are going to do a "name wallet" it might be better to have a pop-up with only "write name for wallet" on it, so user knows exactly what's going on).

- I selected a few mainnet and testnet ( the default mainnet selection is a little bit odd, the default had Gnosis selected only), when I was on wallet screen I saw all the networks I selected (main and testnet). When I did a swipe refresh I could see only mainnet (I had last been selecting mainnet). Recommend that on the network selection with this option selected, we show mainnets then testnets, without any switch (ie all on same listview). When user exits back to the wallet page, it should show the mainnets only.

Once this is working let's update it so that it matches the new spec Tomek designed; this will have to be done carefully though.

**Comment by seabornlee** [comment]
@JamesSmartCell I am adding more unit tests.

**Comment by seabornlee** [comment]
@justindg Please also try to modify e2e test, I added it https://github.com/AlphaWallet/alpha-wallet-android/pull/2751/commits/fed2a83556e53bed413e08b8702187d250b3d11e
You can run it by start emulator and run shell `sh e2e.sh`.

**Comment by justindg** [comment]
> @justindg Please also try to modify e2e test, I added it [fed2a83](https://github.com/AlphaWallet/alpha-wallet-android/commit/fed2a83556e53bed413e08b8702187d250b3d11e) You can run it by start emulator and run shell `sh e2e.sh`.

Thanks Seaborn, yes I will! Still working on the feature

**Comment by justindg** [comment]
> Issues found:
> 
> * After clicking "create New Wallet", app goes to the "name wallet" screen but it's not clear why or what user is supposed to do here (I was initially confused about what was meant to happen here. If we are going to do a "name wallet" it might be better to have a pop-up with only "write name for wallet" on it, so user knows exactly what's going on).
> * I selected a few mainnet and testnet ( the default mainnet selection is a little bit odd, the default had Gnosis selected only), when I was on wallet screen I saw all the networks I selected (main and testnet). When I did a swipe refresh I could see only mainnet (I had last been selecting mainnet). Recommend that on the network selection with this option selected, we show mainnets then testnets, without any switch (ie all on same listview). When user exits back to the wallet page, it should show the mainnets only.
> 
> Once this is working let's update it so that it matches the new spec Tomek designed; this will have to be done carefully though.

@JamesSmartCell Fixed the second issue, but couldn't replicate the first. I checked the code and there are very few references to `NameThisWalletActivity`, is it consistently reproducible on your end?

**Comment by JamesSmartCell** [comment]
@justindg ok It might have been just an artifact but I also can't reproduce where it jumps to the naming screen, so lets leave that for now.

Try this:

1. Create new wallet
2. Select some mainnets.
3. Select some testnets.

Exit selection.

Screen looks like this:

screen

You can fix it very simply (maybe, if you're receiving a return intent) by sending an instruction to refresh the wallet screen after user finishes their selection. Unless you added a custom function into TokensService to update the filter outside of the normal setup. In which case you should remove that refresh TokensService then it'll work correctly.

**Comment by justindg** [comment]
> @justindg ok It might have been just an artifact but I also can't reproduce where it jumps to the naming screen, so lets leave that for now.
> 
> Try this:
> 
> 1. Create new wallet
> 2. Select some mainnets.
> 3. Select some testnets.
> 
> Exit selection.
> 
> Screen looks like this:
> 
> screen
> 
> You can fix it very simply (maybe, if you're receiving a return intent) by sending an instruction to refresh the wallet screen after user finishes their selection. Unless you added a custom function into TokensService to update the filter outside of the normal setup. In which case you should remove that refresh TokensService then it'll work correctly.

This has been fixed

**Comment by seabornlee** [comment]
@JamesSmartCell I am all done on this PR, transfer test is also enabled and it works well on Görli testnet.

**Review by JamesSmartCell — APPROVED** [review]
@justindg works as expected now. Nice.

File: app/build.gradle [hunk 1]
```diff
@@ -312,12 +305,12 @@ dependencies {
     androidTestImplementation 'androidx.test.uiautomator:uiautomator:2.2.0'
     androidTestImplementation 'androidx.test.espresso:espresso-contrib:3.4.0'
 
-    implementation 'com.trustwallet:wallet-core:2.6.4'
+    implementation group: 'com.trustwallet', name: 'wallet-core', version: '2.6.3'
 
-    implementation 'com.github.florent37:TutoShowcase:d8b91be8a2'
+    implementation 'com.github.florent37:tutoshowcase:1.0.1'
```
[the remark this answers is not in the document]
Comment (JamesSmartCell) [reply app/build.gradle:315]: Use 2.6.3 because 2.6.4 requires extra setup

**Pull request merged**